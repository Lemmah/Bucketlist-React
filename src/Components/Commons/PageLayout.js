import React from 'react'
import { Grid, Row, Col, Image } from 'react-bootstrap'
import RegistrationForm from './../Authentication/RegistrationForm'
import Bucketlist from './../Bucketlists/Bucketlist'
import BucketlistItem from './../Bucketlists/BucketlistItem'
import Requests from './../APICalls/Config'


class PageLayout extends Requests{
	constructor(){
		super();
		this.state = {
			token: null,
			username: null,
			bucketlists: null,
			items: null,
			bucketlistOnFocus: null,
			firstFocus: true,
		}

		this.getAllBucketlistItems = this.getAllBucketlistItems.bind(this)
		this.createBucketlist = this.createBucketlist.bind(this)
		this.createBucketlistItem = this.createBucketlistItem.bind(this)
		this.createBucketlistItem = this.createBucketlistItem.bind(this)
		this.deleteBucketlist = this.deleteBucketlist.bind(this)
		this.deleteBucketlistItem = this.deleteBucketlistItem.bind(this)
		this.updateBucketlistItem = this.updateBucketlistItem.bind(this)
		this.updateBucketlist = this.updateBucketlist.bind(this)
	} 

	componentWillReceiveProps(nextProps){
		console.log(nextProps)
		let bucketlists = this.state.bucketlists;
		if(nextProps.newBucketlist){
			if(!bucketlists) {
				bucketlists = []
			}
				bucketlists.splice(0,0,nextProps.newBucketlist);
				this.setState({bucketlists: bucketlists})
		}

		let items = this.state.items;
		if(nextProps.newBucketlistItem){
			if(!items){
				items = []
			}
				items.push(nextProps.newBucketlistItem);
				this.setState({items: items})
		}
		if(nextProps.authenticated){
			if(nextProps.authenticated.message.user
				=== this.state.username){
				return;
			} else {
				const token = nextProps.authenticated.message.access_token;
				this.setState({
					username: nextProps.authenticated.message.user,
					token: token,
				});
				this.getAllBucketlists(token);
			}
		}
	}

	componentDidUpdate(prevProps, prevState){
		if(this.state.bucketlists){
			if(this.state.bucketlists.length > 0){
				if(this.state.firstFocus){
					let firstBucketlist = document.getElementsByClassName("bucketlist")[0];
					(firstBucketlist ? firstBucketlist.click() : "");
					this.setState({
						firstFocus: false,
					});
				}
			}
		}
	}

	registerUser(credentials){
    this.props.onRegister(credentials);
	}

	createBucketlist(resourceUrl, details){
		this.props.onCreateBucketlist(resourceUrl, details);
	}

	createBucketlistItem(resourceUrl, details){
		this.props.onCreateBucketlistItem(resourceUrl, details);
	}

	getAllBucketlists(token){
		const resource_name = "bucketlists";
		const resource_url = "/bucketlists";
		this.getResource(resource_name, resource_url, token);
	}

	getAllBucketlistItems(bucketlist){
		const resource_name = "items";
		const resource_url = "/bucketlists/" + bucketlist.id + "/items";
		this.getResource(resource_name, resource_url, this.state.token);
		this.setState({
      	bucketlistOnFocus: bucketlist,
    });
	}

	deleteBucketlist(id){
		let bucketlists = this.state.bucketlists
		let index = bucketlists.findIndex(bl => bl.id === id)
		bucketlists.splice(index, 1)
		this.setState({
			bucketlists: (bucketlists.length > 0 ? 
										bucketlists : null),
			bucketlistOnFocus: null,
			items: null,
			firstFocus: true,
		})
		const token = this.state.token
		const resourceUrl = "/bucketlists/" + id
		this.deleteResource(resourceUrl, token)
	}

	deleteBucketlistItem(id){
		let items = this.state.items
		let index = items.findIndex(bli => bli.id === id)
		items.splice(index, 1)
		this.setState({
			items: (items.length > 0 ?
							items : null),
		})
		const token = this.state.token
		let bucketlistId = this.state.bucketlistOnFocus.id
		const resourceUrl = "/bucketlists/"+bucketlistId+"/items/"+id
		this.deleteResource(resourceUrl, token)
	}

	updateBucketlistItem(details, itemOnFocus){
		const items = this.state.items
    const itemIndex = items.findIndex(item => item === itemOnFocus)
    let targetItem = items[itemIndex]
    targetItem.name = details
    items.splice(itemIndex, 1)
    items.splice(0,0,targetItem)
    this.setState({
    	items: items,
    })
    const token = this.state.token
    const bucketlistId = this.state.bucketlistOnFocus.id
    const resourceUrl = "/bucketlists/"+bucketlistId+"/items/"+targetItem.id
    const newDetails = { name: details }
    this.updateResource(resourceUrl, newDetails, token)
	}

	updateBucketlist(details){
		const targetBucketlistId = this.state.bucketlistOnFocus.id
		const bucketlists = this.state.bucketlists
		const targetIndex = bucketlists.findIndex(bucketlist => bucketlist.id === targetBucketlistId)
		bucketlists[targetIndex].name = details
		this.setState({
			bucketlists: bucketlists,
		})
		const token = this.state.token
		const resourceUrl = "/bucketlists/"+targetBucketlistId
		const newDetails = { name: details }
		this.updateResource(resourceUrl, newDetails, token)
	}

	rowContent(){
		if(this.props.authenticated){
			return (
				<Row>
		      <Col xs={12} md={4}>
		      	<Bucketlist 
		      		bucketlists={this.state.bucketlists}
		      		token={this.state.token}
		      		getAllBucketlistItems={this.getAllBucketlistItems}
		      		createBucketlist={this.createBucketlist}
		      	/>
		      </Col>
		      <Col xs={12} md={8}>
		      	<BucketlistItem
		      		bucketlist={this.state.bucketlistOnFocus}
		      		items={this.state.items}
		      		createBucketlistItem={this.createBucketlistItem}
		      		deleteBucketlist={this.deleteBucketlist}
		      		deleteBucketlistItem={this.deleteBucketlistItem}
		      		updateBucketlistItem={this.updateBucketlistItem}
		      		updateBucketlist={this.updateBucketlist}
		      	/>
		      </Col>
		    </Row>
			);
		} else {
			return (
				<Row>
			      <Col xs={0} md={8}>
			      	<Image
			      		style={{maxHeight: 460}}
			      		src="dreams.jpg" responsive thumbnail
			      	/>
			      </Col>
			      <Col xs={12} md={4}>
			      	<RegistrationForm
			      		onRegister={this.registerUser.bind(this)}
			      		registrationError={this.props.registrationError}
			      	/>
			      </Col>
			  </Row>
			);
		}
	}

	render(){
		return (
			<section>
				<Grid>
					{this.rowContent()}
			  </Grid>
			</section>
		);
	}
}

export default PageLayout;