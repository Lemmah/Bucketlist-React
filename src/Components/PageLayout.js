import React, { Component } from 'react'
import { Grid, Row, Col, Image } from 'react-bootstrap'
import RegistrationForm from './RegistrationForm'
import axios from 'axios'

const api_url="http://localhost:5000"

class PageLayout extends Component{
	constructor(){
		super();
		this.state = {
			username: null,
			bucketlists: null,
		}
	}

	registerUser(credentials){
    this.props.onRegister(credentials);
	}

	componentWillReceiveProps(nextProps){
		if(nextProps.authenticated){
			console.log("Received")
			if(nextProps.authenticated.message.user
				=== this.state.username){
				console.log("Unchanged");
				return;
			} else {
				const token = nextProps.authenticated.message.access_token;
				this.setState({
					username: nextProps.authenticated.message.user,
				});
				this.getAllBucketlists(token);
				console.log("Changed")
			}
		}
	}

	getAllBucketlists(token){
		axios.get(api_url + "/bucketlists", {
			headers: {
				'Authorization': 'Bearer ' + token,
				'Content-Type': 'application/json',
			}
		})
    .then((response) => {
      console.log(response.data.message);
    })
    .catch((error) => {
      console.log(error);
    });
	}

	rowContent(){
		if(this.props.authenticated){
			console.log("authenticated")
			return (
				<Row>
		      <Col xs={12} md={8}>Authenticated</Col>
		      <Col xs={12} md={4}>...</Col>
		    </Row>
			);
		} else {
			return (
				<Row>
			      <Col xs={0} md={8}>
			      	<Image
			      		style={{maxHeight: 460}}
			      		src="dreams.jpg" responsive thumbnail
			      	/>
			      </Col>
			      <Col xs={12} md={4}>
			      	<RegistrationForm
			      		onRegister={this.registerUser.bind(this)}
			      		registrationError={this.props.registrationError}
			      	/>
			      </Col>
			  </Row>
			);
		}
	}

	render(){
		return (
			<section>
				<Grid>
					{this.rowContent()}
			  </Grid>
			</section>
		);
	}
}

export default PageLayout;