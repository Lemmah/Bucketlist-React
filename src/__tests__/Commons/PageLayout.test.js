import React from 'react';
import renderer from 'react-test-renderer';
import PageLayout from '../../Components/Commons/PageLayout';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
configure({ adapter: new Adapter() });
import { shallow, mount, render } from 'enzyme';

const dummyCredentials = {
	"email" : "jnlemayian@gmail.com",
	"password": "easyPassword", 
}

const dummyAuthenticationInfo = {
    "message": {
        "access_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJleHAiOjE1MDcwMDkyMTcsImlhdCI6MTUwNzAwNTYxNywic3ViIjoxfQ.idObEdx7BfAOeKGmh8w80wk-hx5J-JhUgWbiYDqb1fE",
        "info": "You logged in successfully.",
        "user": "jnlemayian@gmail.com"
    }
}

const sampleBucketlists = [
      {
          "created_by": 1,
          "date_created": "Mon, 02 Oct 2017 20:10:40 GMT",
          "date_modified": "Mon, 02 Oct 2017 20:10:40 GMT",
          "id": 138,
          "name": "Lemmah"
      },
      {
          "created_by": 1,
          "date_created": "Mon, 02 Oct 2017 20:10:48 GMT",
          "date_modified": "Mon, 02 Oct 2017 20:10:48 GMT",
          "id": 139,
          "name": "Lemmah22"
      },
      null
  ]

const sampleItems = [
    {
        "belongs_to": 138,
        "date_created": "Mon, 02 Oct 2017 20:14:10 GMT",
        "date_modified": "Mon, 02 Oct 2017 20:14:10 GMT",
        "id": 119,
        "name": "Lemmah22"
    },
    {
        "belongs_to": 138,
        "date_created": "Mon, 02 Oct 2017 20:14:17 GMT",
        "date_modified": "Mon, 02 Oct 2017 20:14:17 GMT",
        "id": 120,
        "name": "Lemmah"
    }
]

const dummyProps = {
	newBucketlist: sampleBucketlists[1],
	newBucketlistItem: sampleItems[1],
	authenticated: dummyAuthenticationInfo,
	registrationError: null, 
}

const dummyEmptyProps = {
	newBucketlist: null,
	newBucketlistItem: null,
	authenticated: dummyAuthenticationInfo,
	registrationError: null, 
}

it('Displays PageLayout without Crashing', () => {
	const rendered = renderer.create(
		<PageLayout />,
		);
	expect(rendered.toJSON()).toMatchSnapshot();
});

describe('Shows Dashboard if logged in', () => {
	const wrapper = mount(
		<PageLayout
	      newBucketlist={null}
	      newBucketlistItem={null}
	      authenticated={dummyAuthenticationInfo}
	      onRegister={()=> {}}
	      registrationError={()=> {}}
	      onCreateBucketlist={()=> {}}
	      onCreateBucketlistItem={()=> {}}
	  />
	);
	const instance = wrapper.instance();

	it('Does Set state when it Receives Props', () => {
		instance.componentWillReceiveProps(dummyProps);
		instance.componentWillReceiveProps(dummyEmptyProps);
	})

	/* Set mock Mounted state */
	instance.setState({
		token: dummyAuthenticationInfo.message.access_token,
		user: dummyAuthenticationInfo.message.user,
		bucketlists: sampleBucketlists,
		items: sampleItems,
		bucketlistOnFocus: sampleBucketlists[0],
		firstFocus: false,
	});

	it('Registers new User using Credentials', () => {
		instance.registerUser(dummyCredentials);
	});

	it('Creates Bucketlists', () => {
		const resourceUrl = '/bucketlists';
		const details = {
			'name' : 'Travel Round the World',
		}
		instance.createBucketlist(resourceUrl, details);
	})

	it('Creates Bucketlist Items', () => {
		const resourceUrl = '/bucketlists/138/items';
		const details = {
			'name' : 'Travel to Russia',
		}
		instance.createBucketlistItem(resourceUrl, details);
	})

	it('Updates Bucketlist item', () => {
		const details = "Travel to England";
		const itemOnFocus = sampleItems[0];
		instance.updateBucketlistItem(details, itemOnFocus);
	});

	it('Updates Bucketlist', () => {
		const details = "Travelling Round the World";
		instance.updateBucketlist(details);
	});

	it('Gets all Bucketlist items', () =>{
		const bucketlist = sampleBucketlists[0];
		instance.getAllBucketlistItems(bucketlist);
	})

	it('Deletes Bucketlist Item using Id', () => {
		instance.deleteBucketlistItem(119);
	});


	it('Deletes Bucketlist using Id', () => {
		instance.deleteBucketlist(139);
	});


})