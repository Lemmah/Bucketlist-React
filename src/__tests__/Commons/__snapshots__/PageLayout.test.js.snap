// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays PageLayout without Crashing 1`] = `
<section>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-8 col-xs-0"
      >
        <img
          className="img-responsive img-thumbnail"
          src="dreams.jpg"
          style={
            Object {
              "maxHeight": 460,
            }
          }
        />
      </div>
      <div
        className="col-md-4 col-xs-12"
      >
        <form
          onSubmit={[Function]}
        >
          <h3>
             Start Following your Dreams 
          </h3>
          <p>
             It's Good and Free for good! 
          </p>
          <div
            className="form-group has-feedback"
          >
            <div
              className="form-group"
            >
              <span
                className="input-group"
              >
                <span
                  className="input-group-btn"
                >
                  <button
                    className="btn btn-default"
                    disabled={false}
                    style={
                      Object {
                        "maxHeight": 34,
                        "width": 150,
                      }
                    }
                    type="button"
                  >
                    <label
                      className="control-label"
                      htmlFor={undefined}
                    >
                      Username
                    </label>
                  </button>
                </span>
                <input
                  className="form-control"
                  placeholder="Enter Username"
                  required={true}
                  type="text"
                />
              </span>
            </div>
            <div
              className="form-group"
            >
              <span
                className="input-group"
              >
                <span
                  className="input-group-btn"
                >
                  <button
                    className="btn btn-default"
                    disabled={false}
                    style={
                      Object {
                        "maxHeight": 34,
                        "width": 150,
                      }
                    }
                    type="button"
                  >
                    <label
                      className="control-label"
                      htmlFor={undefined}
                    >
                      Email
                    </label>
                  </button>
                </span>
                <input
                  className="form-control"
                  placeholder="youremail@yourdomain.com"
                  required={true}
                  type="email"
                />
              </span>
            </div>
            <div
              className="form-group has-feedback"
            >
              <span
                className="input-group"
              >
                <span
                  className="input-group-btn"
                >
                  <button
                    className="btn btn-default"
                    disabled={false}
                    style={
                      Object {
                        "maxHeight": 34,
                        "width": 150,
                      }
                    }
                    type="button"
                  >
                    <label
                      className="control-label"
                      htmlFor="formBasicText"
                    >
                      Password
                    </label>
                  </button>
                </span>
                <input
                  className="form-control"
                  onChange={[Function]}
                  placeholder="Enter Password"
                  type="password"
                  value=""
                />
              </span>
              <span
                className="help-block"
                style={
                  Object {
                    "fontSize": 10,
                  }
                }
              />
            </div>
            <div
              className="form-group has-feedback"
            >
              <span
                className="input-group"
              >
                <span
                  className="input-group-btn"
                >
                  <button
                    className="btn btn-default"
                    disabled={false}
                    style={
                      Object {
                        "maxHeight": 34,
                        "width": 150,
                      }
                    }
                    type="button"
                  >
                    <label
                      className="control-label"
                      htmlFor="formBasicText"
                    >
                      Confirm Password
                    </label>
                  </button>
                </span>
                <input
                  className="form-control"
                  onChange={[Function]}
                  placeholder="Enter Password"
                  type="password"
                  value=""
                />
              </span>
              <span
                className="help-block"
                style={
                  Object {
                    "fontSize": 10,
                  }
                }
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <span
              className="input-group"
            >
              <div
                className="btn-toolbar"
                role="toolbar"
              >
                <div
                  className="btn-group"
                >
                  <input
                    className="btn btn-success"
                    style={
                      Object {
                        "maxHeight": 34,
                        "width": 150,
                      }
                    }
                    type="submit"
                    value="Signup"
                  />
                </div>
                <div
                  className="btn-group"
                >
                  <button
                    className="btn btn-default"
                    disabled={false}
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m29.4 0.3v5.9h-3.5q-1.9 0-2.6 0.8t-0.7 2.4v4.2h6.6l-0.9 6.6h-5.7v16.9h-6.8v-16.9h-5.7v-6.6h5.7v-4.9q0-4.1 2.3-6.4t6.2-2.3q3.3 0 5.1 0.3z"
                        />
                      </g>
                    </svg>
                  </button>
                  <button
                    className="btn btn-default"
                    disabled={false}
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m37.7 9.1q-1.5 2.2-3.7 3.7 0.1 0.3 0.1 1 0 2.9-0.9 5.8t-2.6 5.5-4.1 4.7-5.7 3.3-7.2 1.2q-6.1 0-11.1-3.3 0.8 0.1 1.7 0.1 5 0 9-3-2.4-0.1-4.2-1.5t-2.6-3.5q0.8 0.1 1.4 0.1 1 0 1.9-0.3-2.5-0.5-4.1-2.5t-1.7-4.6v0q1.5 0.8 3.3 0.9-1.5-1-2.4-2.6t-0.8-3.4q0-2 0.9-3.7 2.7 3.4 6.6 5.4t8.3 2.2q-0.2-0.9-0.2-1.7 0-3 2.1-5.1t5.1-2.1q3.2 0 5.3 2.3 2.4-0.5 4.6-1.7-0.8 2.5-3.2 3.9 2.1-0.2 4.2-1.1z"
                        />
                      </g>
                    </svg>
                  </button>
                  <button
                    className="btn btn-default"
                    disabled={false}
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m20.1 17.5h16.2q0.3 1.5 0.3 2.9 0 4.8-2 8.7t-5.8 5.9-8.7 2.1q-3.5 0-6.6-1.3t-5.5-3.7-3.7-5.4-1.3-6.7 1.3-6.7 3.7-5.4 5.5-3.7 6.6-1.3q6.7 0 11.5 4.4l-4.6 4.5q-2.8-2.6-6.9-2.6-2.8 0-5.3 1.4t-3.9 4-1.4 5.4 1.4 5.4 3.9 4 5.3 1.4q2 0 3.6-0.5t2.7-1.3 1.8-1.9 1.2-1.9 0.5-1.8h-9.8v-5.9z"
                        />
                      </g>
                    </svg>
                  </button>
                  <button
                    className="btn btn-default"
                    disabled={false}
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m20.1 2.9q4.7 0 8.6 2.3t6.3 6.2 2.3 8.6q0 5.6-3.3 10.1t-8.4 6.2q-0.6 0.1-0.9-0.2t-0.3-0.7q0 0 0-1.7t0-3q0-2.1-1.2-3.1 1.3-0.2 2.3-0.4t2.1-0.9 1.8-1.5 1.2-2.3 0.5-3.4q0-2.7-1.8-4.6 0.8-2-0.2-4.5-0.6-0.2-1.8 0.2t-2 1l-0.9 0.5q-2-0.6-4.3-0.6t-4.2 0.6q-0.4-0.2-1-0.6t-1.9-0.8-1.9-0.3q-1 2.5-0.1 4.5-1.8 1.9-1.8 4.6 0 1.9 0.5 3.4t1.1 2.3 1.8 1.5 2.1 0.9 2.3 0.4q-0.9 0.8-1.1 2.3-0.4 0.2-1 0.3t-1.3 0.1-1.4-0.5-1.3-1.4q-0.4-0.7-1-1.1t-1.1-0.6l-0.5 0q-0.5 0-0.6 0.1t-0.1 0.2 0.2 0.3 0.2 0.3l0.2 0.1q0.5 0.2 1 0.9t0.7 1.1l0.2 0.5q0.3 0.9 1 1.4t1.5 0.7 1.5 0.1 1.3-0.1l0.5 0q0 0.8 0 1.9t0 1.2q0 0.5-0.3 0.7t-0.9 0.2q-5.2-1.7-8.4-6.2t-3.3-10.1q0-4.7 2.3-8.6t6.2-6.2 8.6-2.3z m-10.6 24.6q0.1-0.2-0.2-0.3-0.2-0.1-0.2 0.1-0.1 0.1 0.1 0.2 0.2 0.2 0.3 0z m0.7 0.7q0.1-0.1-0.1-0.3-0.2-0.2-0.3-0.1-0.2 0.1 0 0.4 0.3 0.2 0.4 0z m0.7 1q0.2-0.1 0-0.4-0.2-0.3-0.4-0.1-0.2 0.1 0 0.4t0.4 0.1z m0.9 1q0.2-0.2-0.1-0.4-0.3-0.3-0.4-0.1-0.2 0.2 0 0.4 0.3 0.3 0.5 0.1z m1.3 0.5q0-0.2-0.3-0.3-0.4-0.1-0.4 0.1t0.2 0.4q0.4 0.1 0.5-0.2z m1.4 0.1q0-0.2-0.4-0.2-0.4 0-0.4 0.2 0 0.3 0.4 0.3 0.4 0 0.4-0.3z m1.3-0.2q-0.1-0.2-0.4-0.2-0.4 0.1-0.3 0.4t0.4 0.1 0.3-0.3z"
                        />
                      </g>
                    </svg>
                  </button>
                  <button
                    className="btn btn-default"
                    disabled={false}
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m10.8 13.9v22.2h-7.4v-22.1h7.4z m0.5-6.8q0 1.7-1.2 2.7t-3 1.1h0q-1.9 0-3-1.1t-1.1-2.7q0-1.6 1.2-2.7t3-1.1 2.9 1.1 1.2 2.7z m26 16.3v12.7h-7.4v-11.9q0-2.3-0.9-3.6t-2.8-1.4q-1.4 0-2.3 0.8t-1.5 1.9q-0.2 0.7-0.2 1.8v12.4h-7.3q0-8.9 0-14.5t0-6.6l0-1h7.3v3.2h-0.1q0.5-0.7 1-1.3t1.2-1.1 2-1 2.5-0.4q3.8 0 6.2 2.6t2.3 7.4z"
                        />
                      </g>
                    </svg>
                  </button>
                </div>
              </div>
            </span>
            <span
              className="help-block"
            >
              - Login or Signup with your social network by clicking on it.
            </span>
            <span
              className="help-block"
            >
              - By signing up, you agree to our terms and condtions.
            </span>
          </div>
        </form>
      </div>
    </div>
  </div>
</section>
`;
